src/guide-test-fn-rng.md

```diff
@@ -0,0 +1,101 @@
+# Testing functions that use RNGs
+
+Occasionally a function that uses random number generators might need to be tested. For functions that need to be tested with test vectors, the following approach might be adapted:
+
+```rust
+use rand::{RngCore, CryptoRng, rngs::OsRng};
+
+pub struct CryptoOperations<R: RngCore + CryptoRng + Default = OsRng> {
+    rng: R
+}
+
+impl<R: RngCore + CryptoRng + Default> CryptoOperations<R> {
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            rng: R::default()
+        }
+    }
+
+    pub fn xor_with_random_bytes(&mut self, secret: &mut [u8; 8]) -> [u8; 8] {
+        let mut mask: [u8; 8] = [0; 8];
+        self.rng.fill_bytes(&mut mask);
+
+        for (byte, mask_byte) in secret.iter_mut().zip(mask.iter()) {
+            *byte ^= mask_byte;
+        }
+
+        mask
+    }
+}
+
+fn main() {
+    let mut crypto_ops = <CryptoOperations>::new();
+
+    let mut secret: [u8; 8] = *b"\x00\x01\x02\x03\x04\x05\x06\x07";
+    let mask = crypto_ops.xor_with_random_bytes(&mut secret);
+    
+    println!("Modified Secret (XORed): {:?}", secret);
+    println!("Mask: {:?}", mask);
+}
+```
+
+And as for tests, we can create a MockRng that implements RngCore and CryptoRng and provide a Default implementation with the value we want to return:
+
+```rust
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[derive(Clone, Copy, Debug)]
+    struct MockRng {
```

Comment:
I suggest renaming to `MockCryptoRng` to make it abundantly clear that this is a mock RNG for testing cryptographic code — something you do not want to have mis-used!

Reply:
Fixed.